src/WorkflowCore/Services/BackgroundTasks/EventConsumer.cs

```diff
@@ -36,6 +36,7 @@ protected override async Task ProcessItem(string itemId, CancellationToken cance
             if (!await _lockProvider.AcquireLock($"evt:{itemId}", cancellationToken))
             {
                 Logger.LogInformation($"Event locked {itemId}");
+                await QueueProvider.QueueWork(itemId, QueueType.Event);
```

Comment:
Could this cause a [poison message](https://en.wikipedia.org/wiki/Poison_message)?

Reply:
This line may be. Actually it looks like adding to a queue when failing to acqure an evt lock is a bad idea. But it is necessary when failing to acquire a workflow lock.